monitor/internal/pod/resync.go

```diff
@@ -35,5 +55,75 @@ func ResyncWithAllPods(ctx context.Context, c client.Client, evCh chan<- event.G
 		}
 	}
 
+	// now wait for all pods to have reported back
+waitLoop:
+	for {
```

Comment:
We can slightly improve this part ?

```
j:=0
for {
	select{
	case info := <-*i.GetInfoCh():
		if _, ok := m[info]; !ok {
			continue
		}
		j++
	default:
		if j==len(m){
			i.DisableNeedsInfo()
			return nil
		}
	}
}		
```

Reply:
We can make m = `map[string]struct{}`
No need to use labels.
No need for internal for loop.

Reply:
this doesn't work unfortunately ... I'm filling the map first to make sure these are the ones that I want to look out for. And there is also no guarantee that I get only _one_ event for every pod. On the contrary, there are usually 4-5 reconciles for one pod starting up. So I have to track them all on its own.